### ISSUE TITLE
Selenium Grid doesn't pass chromeOptions through to browser when Capabilities are provided programmatically AND the Selenium Grid Node has a config file.

### ISSUE THREAD

**Issue by stephenkuhn214** [SeleniumHQ/selenium#7084]
When chromeOptions are set via the Selenium-Webdriver API, the browser receives the argument and handles it accordingly.  However when the Selenium Grid Node is started with the -nodeConfig argument, these chromeOptions (being passed the same way) are being ignored.

**Driver Initialization**

    let driver = new Builder();

    driver = driver.usingServer(this.seleniumAddress);

    var chromeOptions = new Chrome.Options();
    chromeOptions.addArguments('--start-maximized');
    chromeOptions.setChromeBinaryPath(this.browserBinaryPath);

    driver = await driver.setChromeOptions(chromeOptions);

    driver = await driver.forBrowser(this.browserName, this.browserVersion).build();

**SeleniumNodeConfig.json**

    {
      "capabilities": [
    	{
          "browserName": "chrome",
          "version": "",
          "platform": "WINDOWS",
          "maxInstances": 1
        },
        {
          "browserName": "firefox",
          "version": "",
          "platform": "WINDOWS",
          "maxInstances": 1
        },
        {
          "browserName": "internet explorer",
          "version": "11",
          "platform": "WINDOWS",
          "maxInstances": 1
        }
      ]
    }

**Starting Selenium Node**

    java -jar <path>\selenium-server-standalone-3.141.5.jar -role node -hub %hubURL% -nodeConfig SeleniumNodeConfig.json

If I start the node without the NodeConfig file, the browser starts maximized, as it should, and I see this in my terminal output:

    18:08:17.330 INFO [ActiveSessionFactory.apply] - Capabilities are: {
      "browserName": "chrome",
      "chromeOptions": {
        "args": [
          "--start-maximized"
        ],
        "binary": "C:\\Program Files (x86)\\Google\\Chrome\\Application\\chrome.exe"
      }
    }

But when I add the node config back in, it restricts the browser instances to 1 as it should, but not longer will start maximized.

I've tried adding 

	  "chromeOptions": {
		"args": ["--start-maximized"]
	  }

and

	  "goog:chromeOptions": {
		"args": ["--start-maximized"]
	  }

to my `SeleniumNodeConfig.json` file, but those don't even show up in the terminal when the node starts the session.

### Environment

OS: Selenium Hub on Linux, Node on Windows 7
Browser: Chrome
Browser version: 73.0.3683.86
Browser Driver version: 2.46.628402
Selenium Grid version (if applicable): 3.141.5

**Comment by cgoldberg** [comment]
could you be hitting this?  https://bugs.chromium.org/p/chromedriver/issues/detail?id=1855

**Comment by barancev** [comment]
I can't reproduce this issue with Java binding. Seems to be a nodejs-specific issue

**Comment by barancev** [comment]
And it seems to be fixed already, I can't reproduce it with the nodejs binding built from the HEAD. So it's matter of the next release date.

**Comment by Anedumgottil** [comment]
I'm still running into this issue, is anyone else?

**Comment by lorenzodallavecchia** [comment]
I have just run into part of this issue. The selenium version is 3.141.59.
I was trying to set some default chromeOptions in the node configuration file, so as to avoid passing them from every client that uses the Grid.

This is what I tried to do in my node config JSON.

```json
{
  "capabilities": [
    {
      "browserName": "chrome",
      "maxInstances": 5,
      "seleniumProtocol": "WebDriver",
      "chromeOptions": { "w3c": false }
    }
  ]
}
```

The `chromeOptions` part is completely ignored: it does not show up in the console and does not have the desired effect. I also tried with the new `goog:chromeOptions` as some suggested: no good.

I looked into the Selenium code and found that it uses only the Chrome Options coming from the client request. The only place where some options are mixed-in from the configuration is [ChromeMutator](/SeleniumHQ/selenium/blob/selenium-3.141.59/java/server/src/org/openqa/grid/selenium/node/ChromeMutator.java), but it only replaces the specific property `chrome_binary`.

So it looks like Selenium 3.x completely ignores `chromeOptions` specified in the JSON file.

However I see that a major refactoring is coming in version 4. Can we expect this to be supported then?
